py/objtype.c

```diff
@@ -542,7 +542,18 @@ const byte mp_binary_op_method_name[MP_BINARY_OP_NUM_RUNTIME] = {
 };
 
 static mp_obj_t instance_binary_op(mp_binary_op_t op, mp_obj_t lhs_in, mp_obj_t rhs_in) {
+    #if MICROPY_PY_TSTRINGS
+    if (op == MP_BINARY_OP_ADD || op == MP_BINARY_OP_INPLACE_ADD) {
```

Comment:
This whole bit can be removed, it's not needed.  There is already a check in `template_binary_op()`.